src/components/progress-circle/progress_circle.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Component,
+  ViewEncapsulation,
+  ElementRef,
+  Attribute,
+  Input,
+  HostBinding
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+
+/** Display modes of Progress Circle */
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+}
+
+
+/**
+ * <md-progress-circle> component.
+ */
+@Component({
+  selector: 'md-progress-circle',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-circle/progress_circle.html',
+  styleUrls: ['./components/progress-circle/progress-circle.css'],
+})
+export class MdProgressCircle {
+  /**
+   * Value of the progress circle.
+   *
+   * Input:number, defaults to 0.
+   * value_ is bound to the host as the attribute aria-valuenow.
+   */
+  @Input('value')
+  value_: number = 0;
+  @HostBinding('attr.aria-valuenow')
+  get _value() {
+    return this.value_;
+  }
+
+  /**
+   * Mode of the progress circle
+   *
+   * Input must be one of the values from ProgressMode, defaults to 'determinate'.
+   * mode is bound to the host as the attribute host.
+   */
+  @Input()
+  @OneOf([ProgressMode.DETERMINATE, ProgressMode.INDETERMINATE])
+  mode: string;
+  @HostBinding('attr.mode')
```

Comment:
From my understanding if we do not do a HostBinding here, then if the mode is changed by the component it will not send out the change without the host binding.   For instance, when an attempt is made to update to an invalid progress mode it changes it to a valid mode (determinate, specifically)